firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3798,11 +3798,12 @@ extension BrowserViewController: TabManagerDelegate {
             webView.accessibilityIdentifier = "contentView"
             webView.accessibilityElementsHidden = false
 
-            browserDelegate?.show(webView: webView)
             if selectedTab.isFxHomeTab {
                 // Added as initial fix for WKWebView memory leak. Needs further investigation.
                 // See: [FXIOS-10612] + [FXIOS-10335]
                 needsReload = true
+            } else {
+                browserDelegate?.show(webView: webView)
```

Comment:
I moved here in the else case, since i don't think that if the `selectedTab` is the homepage we need to show the `webView` but i believe i'm missing something here

Reply:
IMO that part of the code is a very brittle (reloading webview), and I'm a bit nervous to change logic without good reasons to.